### ISSUE TITLE
Recommend config has any[] type

### ISSUE THREAD

**Issue by LuckyWindsck** [intlify/eslint-plugin-vue-i18n#682]
<img width="397" height="101" alt="Image" src="https://github.com/user-attachments/assets/e8522aad-701f-40de-b47c-100a38dfb1da" />

**Tell us about your environment**

- **ESLint version:** v9.39.2
- **eslint-plugin-vue version:** 10.6.2
- **@antfu/eslint-config version:** 6.7.3
- **eslint-plugin-vue-i18n version:** 4.1.0
- **Node version:** v24.12.0

**Please show your full configuration:**

```ts
import vueI18n from '@intlify/eslint-plugin-vue-i18n'

type T = typeof vueI18n.configs.recommended // any[]
```

**What did you do?**

1. Try to spread `...vueI18n.configs.recommended` to anftu eslint config.
2. Got [ts/no-unsafe-argument](https://typescript-eslint.io/rules/no-unsafe-argument) error.

<img width="689" height="134" alt="Image" src="https://github.com/user-attachments/assets/c5e7a9bf-3740-4004-87b9-5781a8fa0349" />

**What did you expect to happen?**

Not `any[]` but exact type

**What actually happened?**

`typeof vueI18n.configs.recommended` is `any[]`

**Repository to reproduce this issue**

None.

`pnpm add -D @intlify/eslint-plugin-vue-i18n@4.1.0` should reproduce it.

https://www.npmjs.com/package/@intlify/eslint-plugin-vue-i18n/v/4.1.0?activeTab=code

<img width="572" height="1117" alt="Image" src="https://github.com/user-attachments/assets/4bb4e565-4edb-4227-9c3c-fd264b797816" />
